SNLS_NLSPowell.h

```diff
@@ -0,0 +1,729 @@
+#pragma once
+
+#include "SNLS_base.h"
+#include "SNLS_qr_solve.h"
+#include "SNLS_linalg.h"
+
+#include <stdlib.h>
+#include <iostream>
+#ifdef __cuda_host_only__
+#include <string>
+#include <sstream>
+#include <iomanip>
+#endif
+
+namespace snls {
+
+/** Helper templates to ensure compliant CRJ implementations */
+template<typename CRJ, typename = void>
+struct nls_has_valid_computeRJ : std::false_type { static constexpr bool value = false;};
```

Comment:
This is duplicative with has_valid_computeRJ in SNLS_TrDLDenseG.h. Unifying would enforce the idea that one can switch among solvers for a given problem. 

Same for nls_has_valid_computeRJ and other helper templates below.